infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/persistencestrategy/counter/ClusteredCounterHandler.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispan.persistencestrategy.counter;
+
+import java.lang.invoke.MethodHandles;
+
+import org.hibernate.ogm.datastore.infinispan.logging.impl.Log;
+import org.hibernate.ogm.datastore.infinispan.logging.impl.LoggerFactory;
+import org.hibernate.ogm.dialect.spi.NextValueRequest;
+
+import org.infinispan.counter.api.CounterConfiguration;
+import org.infinispan.counter.api.CounterManager;
+import org.infinispan.counter.api.CounterType;
+import org.infinispan.counter.api.Storage;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+/**
+ * Provides access to Infinispan Clustered Counter feature.
+ * Used by the dialect to implement a reliable ID generator.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public abstract class ClusteredCounterHandler {
+
+	private static final Log LOG = LoggerFactory.make( MethodHandles.lookup() );
+	protected final EmbeddedCacheManager cacheManager;
+
+	public ClusteredCounterHandler(EmbeddedCacheManager cacheManager) {
+		this.cacheManager = cacheManager;
+		validate();
+	}
+
+	private void validate() {
+		if ( cacheManager.getTransport() == null ) {
+			throw LOG.counterCannotBeCreatedForLocalCaches();
+		}
+	}
+
+	/**
+	 * The method creates the counter only if the counter is not already created.
+	 *
+	 * @param initialValue		initial value for the counter
+	 * @param counterManager	counterManager service
+	 * @param counterName		unique name for the counter
+	 * @return					true if the counter is created by current thread, false otherwise
+	 */
+	protected boolean defineCounterIfNotExists(int initialValue, CounterManager counterManager, String counterName) {
+
+		// if counter is already present, skipping defining
+		if ( counterManager.isDefined( counterName ) ) {
+
+			if ( LOG.isTraceEnabled() ) {
+				LOG.tracev( "Counter {0} is already defined on Infinispan data store: Skip creation", counterName );
+			}
```

Comment:
In this case it doesn't matter.

More critical: I'm not sure if that "{0}" syntax works with JBoss Logger?

Reply:
Yes I think that you're right. I think that isTranceEnabled should give no benefit in term of performance. Maybe it would have given it if we had linked the String log message with +. 
I will replace "{0}" with %1$s.
Thank you